### PR TITLE
[MEDIA EDITOR] added "saturation" filter and do permanent saving on e…

### PR DESCRIPTION
**Pull request by jerome-poisson**
…ach API call

SDESK-3010

### CODE REVIEW

File: superdesk/media/media_editor.py [hunk 1]
```diff
@@ -72,52 +71,62 @@ def transform(self, im, operation, param):
...
         elif operation == 'grayscale':
             return im.convert('L')
+        elif operation == 'saturation':
+            return ImageEnhance.Color(im).enhance(float(param))
         logger.warning('unhandled operation: {operation} {param}'.format(
             operation=operation,
             param=param))
         return im
 
     def create(self, docs):
```
Comment (petrjasek) [inline superdesk/media/media_editor.py]: I this this could be put into a few methods, it's kind of hard to follow what it does

Comment (jerome-poisson) [reply superdesk/media/media_editor.py]: for now it's only calling one method by operation, I could put in separate methods, but that would be on line each time. Do you think it's would really improve readability ?

File: superdesk/media/media_editor.py [hunk 2]
```diff
@@ -61,63 +60,97 @@ def transform(self, im, operation, param):
...
+                                             current_app.media.url_for_media,
+                                             temporary=True)
+
+            # we update item in db, and the item we'll return to client
+            updates = {'renditions': renditions}
+            ids.append(item_id)
+            archive.update(item_id, updates, item)
+            item.update(updates)
+
+            # we can now delete old media
```
Comment (petrjasek) [inline superdesk/media/media_editor.py]: it should be probably kept so it's visible in previous versions

**Comment by jerome-poisson** [comment]
@petrjasek I'm not sure to get it right, there is not more references to those renditions, were should I link them ?

**Comment by jerome-poisson** [comment]
and AFAIK history is not requested.

File: superdesk/media/media_editor.py [hunk 3]
```diff
@@ -61,63 +60,97 @@ def transform(self, im, operation, param):
...
+            for operation, param in edit.items():
+                try:
+                    out = self.transform(out, operation, param)
+                except ValueError as e:
+                    raise errors.SuperdeskApiError.badRequestError('invalid edit instructions: {msg}'.format(msg=e))
+            buf = BytesIO()
+            out.save(buf, format=im.format)
+
+            # we set metadata
+            buf.seek(0)
```
Comment (petr) [inline superdesk/media/media_editor.py]: May be it's better to hide in specific function ? like 
```
def set_metadata
```
And you make code more readeable than. And of course you don't need to write this comment

**Comment by petrjasek** [comment]
I mean to keep the links working for previous item versions

File: superdesk/media/media_editor.py (hunk 3 above)
Comment (jerome-poisson) [reply superdesk/media/media_editor.py]: Hi, thanks for your feedback.
Your suggestion make sense, but I'm not really fond of using methods for 3 lines, I find it more readable to see what's happening directly without having to jump in the code.
Question of taste I guess.
